studioclass-frontend/src/hooks/useMySubscriptions.js

```diff
@@ -6,15 +6,24 @@ const useMySubscriptions = (setError) => {
   const client = useApolloClient();
 
   const updatePostCache = (postAdded) => {
-    const included = (store, post) => store.map(p => p.id).includes(post.id);
+    const included = (store, post) => store.map(p => p?.node.id).includes(post?.node.id);
     
     try {
       const allPostData = client.readQuery({ query: GET_POSTS });
-
-      if (!included(allPostData.posts, postAdded)) {
+      console.log(allPostData);
+      console.log(postAdded);
+      if (!included(allPostData.posts?.edges, postAdded)) {
+        console.log('here');
         client.writeQuery({
           query: GET_POSTS,
-          data: { posts: allPostData.posts.concat(postAdded) }
+          data: {
```

Comment:
Careful of subscriptions. Apollo reconciles the cache via ID; subscriptions can get pretty heavy on the main thread. Something to watch as your app grows.